addons/account_check_printing/models/account_move.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import models, fields
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    property_payment_method_id = fields.Many2one(
```

Comment:
This is not a property.
I don't know if it inherits the `company_dependent` flag from the partner (cc @rco-odoo) but as an `account.move` is always bound to a company, there is no need to make it a property. It should be a real m2o.

Reply:
No need to check with RCO, I changed it

Reply:
This is not a company-dependent field, for the reason @KangOl mentioned.
You should rename it as `payment_method_id` to make this less confusing.

The field definition has more severe issues:
* it should not be recomputed if you modify the partner's `property_payment_method_id` (it depends on `partner_id` and nothing else);
* computing as a related field is wrong if the context's main company is different from the move's.

I suggest to compute it with something like
```python
@api.depends('partner_id')
def _compute_payment_method_id(self):
    for move in self:
        partner = move.partner_id
        # take the payment method corresponding to the move's company
        move.payment_method_id = partner.with_company(move.company_id).property_payment_method_id
```

Reply:
Done